### PR TITLE
Feature/bael 5307 call graphql

### PR DESCRIPTION
**Pull request by dstr89**

### CODE REVIEW

File: core-java-modules/core-java-graphql/.gitignore [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+# Files generated by integration tests
+*.txt
+backup-pom.xml
+/bin/
+/temp
+
+#IntelliJ specific
+.idea/
+*.iml
```
Comment (andrzejsydor) [inline core-java-modules/core-java-graphql/.gitignore:26]: No new line at the end

File: core-java-modules/core-java-graphql/pom.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            <plugin>
+                <artifactId>maven-war-plugin</artifactId>
+                <version>3.1.0</version>
+            </plugin>
+        </plugins>
+    </build>
+
+    <properties>
+        <maven-compiler-plugin.version>3.1</maven-compiler-plugin.version>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (andrzejsydor) [inline core-java-modules/core-java-graphql/pom.xml:93]: Why so old java version?

File: core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/client/apache/ApacheHttpClient.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class ApacheHttpClient {
```
Comment (andrzejsydor) [inline core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/client/apache/ApacheHttpClient.java:14]: could be `final`

File: core-java-modules/core-java-graphql/pom.xml (hunk 2 above)
Comment (ashleyfrieze) [reply core-java-modules/core-java-graphql/pom.xml:136]: Baeldung defaults to support for the earliest possible Java for the examples.

File: core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/Constants.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.baeldung.graphql;
+
+public class Constants {
+
+    public static final String QUERY_PARAMETER = "query";
+
+    private Constants() {}
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/Constants.java:7]: Let's remove this boilerplate. While we might try to appease code scanning tools (which would also want the class to be `final` in this case) for these articles, less is more.

File: core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/clients/AmericanExpressNodes.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import com.baeldung.graphql.Constants;
+import com.baeldung.graphql.data.Data;
+import io.aexp.nodes.graphql.GraphQLRequestEntity;
+import io.aexp.nodes.graphql.GraphQLResponseEntity;
+import io.aexp.nodes.graphql.GraphQLTemplate;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+
+public final class AmericanExpressNodes {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/clients/AmericanExpressNodes.java:12]: Let's ditch the `final` and `private` constructor here too - they're unnecessary noise.

File: core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/clients/ApacheHttpClient.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public final class ApacheHttpClient {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/clients/ApacheHttpClient.java:14]: And here

File: core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/data/BookRepository.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class BookRepository {
+
+    private static List<Book> books = null;
+
+    public BookRepository() {
+        if (books == null) {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/data/BookRepository.java:12]: Oooh - we've got a constructor of an instance object that's actually doing initialization of a static member. That's a bit confusing. Let's do this in the initializer of the field itself:

```java
private static List<Book> books = Stream.of(
    new Book(...), 
    new Book(...))
  .collect(toList()));
```

File: core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/data/Data.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
+package com.baeldung.graphql.data;
+
+import java.util.Collections;
+import java.util.List;
+
+public class Data {
+
+    private List<Book> allBooks;
+
+    public Data() {}
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/data/Data.java:10]: Let's put a line break in these one-line constructors.

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

**Pull request closed without merging**